### PR TITLE
Add/simplify primitives

### PR DESCRIPTION
**Pull request by odersky**
Harmonize trees with numeric types to have the same type. Dotty's alternative to Scala 2's concept of weak conformance. Review by @DarkDimius

### CODE REVIEW

File: src/dotty/tools/dotc/core/SymDenotations.scala [hunk 1]
```diff
@@ -386,6 +386,9 @@ object SymDenotations {
     /** Is symbol a primitive value class? */
     def isPrimitiveValueClass(implicit ctx: Context) = defn.ScalaValueClasses contains symbol
 
+    /** Is symbol a primitive value class? */
+    def isNumericValueClass(implicit ctx: Context) = defn.ScalaNumericValueClasses contains symbol
```
Comment (DarkDimius) [inline src/dotty/tools/dotc/core/SymDenotations.scala]: Naming of this method is not in line with comment. As comment says nothing about class being numeric, and name of the method does not say anything about class being primitive. Methods tests for both of those.

**Comment by DarkDimius** [comment]
Otherwise LGTM.

**Comment by odersky** [comment]
Added comments and rebased to master.

**Pull request merged**